src/main/java/mate/academy/hibernate/relations/dao/impl/ActorDaoImpl.java

```diff
@@ -12,11 +15,50 @@ public ActorDaoImpl(SessionFactory sessionFactory) {
 
     @Override
     public Actor add(Actor actor) {
-        return null;
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = factory.openSession();
+
+            transaction = session.beginTransaction();
+            session.save(actor);
+            transaction.commit();
+
+            return actor;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can't add an actor to DB "
+                    + actor.toString(), e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
```

Comment:
As I know openSesion can't return null it can throw HibernateException which is RuntimeException

Reply:
But i beleive it from the lesson. So I have to check it. Might be good.

Reply:
I guess what you were trying to say is that we catch usual `Exception` here and not `Runtime`? if yes - `RuntimeException` is an instance of `Exception`